control-plane/agents/core/src/node/specs.rs

```diff
@@ -73,4 +74,70 @@ impl ResourceSpecsLocked {
             .map(|n| n.lock().clone())
             .collect()
     }
+
+    /// Cordon the node with the given ID.
+    /// Return the NodeSpec after cordoning.
+    pub(crate) async fn cordon_node(
+        &self,
+        registry: &Registry,
+        node_id: &NodeId,
+        label: String,
+    ) -> Result<NodeSpec, SvcError> {
+        let node = self.get_locked_node(node_id)?;
+        let cordoned_node_spec = {
+            let mut locked_node = node.lock();
+            // Do not allow the same label to be applied more than once.
+            if locked_node.cordon_labels().contains(&label) {
+                return Err(SvcError::CordonLabel {
+                    node_id: node_id.to_string(),
+                    label,
+                });
+            }
+            locked_node.cordon(label);
+            locked_node.clone()
+        };
+        registry.store_obj(&cordoned_node_spec).await?;
```

Comment:
I wonder if node specs should now follow the same pattern we have for the other resources?

Reply:
I think you are probably correct but I'll leave this to a subsequent change.

Reply:
Yep, agreed!